### DISCUSSION TITLE
Edit segment timestamp

### DISCUSSION THREAD

**Discussion by robinmattern** [mifi/lossless-cut#1307] [General]
First let me say, what a fantastic program you have created.  I have used dozens on video authoring programs, but none give me the control over setting cue points that your program does.  Your UI is great, but more importantly I can export and import various timecode files.  

As a longtime JavaScript programmer, I have found so many NPM modules that help me develop my web applications.  But I never thought to look for a video editing tool.  I have just found heaven. 

I will start contributing to your efforts since it seems you are actively making updates.  But here is my first question/suggestion.  Is there are way to set/edit the segment timecode by entering a value?  Not to be critical, but I've found that the current play head position changes erratically after pressing one of the cursor movement keys.  It would be fantastic to be able to visually set a rough begin/end point of a segment, and then fine tune it by moving it with the single frame keys.

**Comment by robinmattern** [comment]
So I'm understanding the dependence on Keyframes.  When I use {ALT-LEFT} and {ALT-RIGHT}, the cue-point start and/or end value will be be the same as the nearest Keyframe value.  So letting the user enter a cue-point value doesn't really make sense. 

I've also found that if I press {I} when sitting on a Keyframe point in Segment {N}, the start point will move left/forward to the first Key point in Segment {N}.  And when I press {O} when sitting in the deleted space just created, the end point of Segment {N-1} will move left/forward to the last Key point in Segment {N-1}.  Voila I have shifted end/start points of the two segments.  

While reading through the "commits" for v3.45.0, I noticed "implement option to shift segments start/end/both".  I see there is a menu command "Edit -> Segments -> Shift all segments on timeline", but is there an command to shift the end and start point of two adjacent segments.  It would really be a "segment shift", since the start and end of the adjacent segments would remain the same.

**Reply by robinmattern** [reply]
I just read through the README file on GitHub.  It mentions "Manual input of cuepoint times".  Where is the this?  Is it "Seek to timecode {G}" followed by "{I} / {O}" or "Split segment at cursor {B}"

**Reply by mifi** [reply]
You can use the text boxes at the bottom to set input output times of the current segment:

<img width="341" alt="Screenshot 2022-10-14 at 14 46 05" src="https://user-images.githubusercontent.com/402547/195850319-a82e0892-62c8-422f-b0b4-d184b61db841.png">

**Comment by robinmattern** [comment]
... analyzing how the "current play head position changes erratically after pressing one of the cursor movement keys", I see that it actually happens whenever the video stops.  The playhead will jump to some apparently random point, and not a keyframe point that I might expect.  Why is this?

**Reply by mifi** [reply]
Not sure what you mean by this, maybe a video will help explain? When the playback is paused, the playhead will jump to the current video playback position where it paused
